packages/back-end/src/services/experiments.ts

```diff
@@ -737,6 +742,13 @@ export async function processPastExperiments(
         el.users += e.users;
         el.numVariations++;
       }
+      if (
+        e.variation_name != e.variation_id &&
+        e.variation_name &&
+        !el.variationNames?.includes(e.variation_name)
+      ) {
+        el.variationNames?.push(e.variation_name);
+      }
```

Comment:
Instead of a separate `if` block, merge this into the one above.  So the code will look like this:

```ts
if (!el.variationKeys.includes(e.variation_id)) {
  el.variationKeys.push(e.variation_id);
  el.weights.push(e.users);
  el.users += e.users;
  el.numVariations++;
  // Below line added
  el.variationNames.push(e.variation_name || "");
}
```

The if block above is checking if this is a new variation that we haven't seen yet.  If so, we do a few things.
1.  Add the variation id to the list of keys
2.  Add the number of users to the list of variation weights
3.  Keep a running sum of the total number of users
4.  Increment the number of variations that we've seen
     
     And now, we're just adding a 5th thing:
6.  Add the variation name to the list of names

Reply:
So, can we assume that there will be the same number of variation names as the number of variation keys? Hence, a name per key?